packages/itwinui-react/src/core/LabeledInput/LabeledInput.tsx

```diff
@@ -104,25 +105,24 @@ export const LabeledInput = React.forwardRef((props, ref) => {
         </Label>
       )}
 
-      <InputWithDecorations
-        status={status}
-        isDisabled={disabled}
+      <Box
+        className={cx('iui-input-with-icon', inputWrapperProps?.className)}
         {...inputWrapperProps}
       >
-        <InputWithDecorations.Input
+        <Input
           disabled={disabled}
           required={required}
+          status={status}
           id={id}
           ref={ref}
           {...rest}
         />
         {shouldShowIcon && (
-          <Icon fill={!svgIcon ? status : undefined} padded {...iconProps}>
+          <EndIcon status={status} {...iconProps}>
```

Comment:
I feel like this component doesn't add anything valuable?

It's the same as:
```tsx
<Icon status={status} {...iconProps} className={cx('iui-end-icon', iconProps?.className)}>
```

Reply:
Ohh, yes, `EndIcon` seems unnecessary; removed (https://github.com/iTwin/iTwinUI/pull/1753/commits/94cf518e3aaa19a08636c25376870dc0c1ee3823).

I think I had it that way since I was basing it off of [`ComboBoxEndIcon`](https://github.com/iTwin/iTwinUI/blob/5d8986451e10101085c98a084fcc4a67b0d76d1e/packages/itwinui-react/src/core/ComboBox/ComboBoxEndIcon.tsx#L16-L36) when I first started working on this PR.